config/config.go

```diff
@@ -168,6 +168,14 @@ type validatable interface {
 	Validate() error
 }
 
+// Unmarshable defines the optional interface for the configuration unmarshaling.
```

Comment:
We should mention what is the default if not implemented. Configs will still be unmarshalled but using the default way. Maybe name the interface `CustomUnmarshable` (not sure asking for opinion)?

Reply:
> Maybe name the interface `CustomUnmarshable` (not sure asking for opinion)?

I agree, that name is less confusing. I will also update the docs to note that there is a default unmarshaler.